src/load/azext_load/data_plane/utils/utils.py

```diff
@@ -0,0 +1,380 @@
+                data = yaml.safe_load(file)
+                if "displayName" in data:
+                    new_body["displayName"] = data["displayName"]
+                if "description" in data:
+                    new_body["description"] = data["description"]
+
+                new_body["keyvaultReferenceIdentityType"] = IdentityType[
+                    "SystemAssigned"
+                ]
+                if "keyvaultReferenceIdentityId" in data:
+                    new_body["keyvaultReferenceIdentityId"] = data[
+                        "keyvaultReferenceIdentityId"
+                    ]
+                    new_body["keyvaultReferenceIdentityType"] = IdentityType[
+                        "UserAssigned"
+                    ]
+
+                if "subnetId" in data:
+                    new_body["subnetId"] = data["subnetId"]
+
+                new_body["loadTestConfiguration"] = body.get(
+                    "loadTestConfiguration", {}
+                )
+                new_body["loadTestConfiguration"]["engineInstances"] = data.get(
+                    "engineInstances", 1
+                )
+                if data.get("certificate"):
+                    new_body["certificate"] = parse_cert(data.get("certificate"))
+                if data.get("secrets"):
+                    new_body["secrets"] = parse_secrets(data.get("secrets"))
+                if data.get("env"):
+                    new_body["environmentVariables"] = parse_env(data.get("env"))
+                # quick test and split csv not supported currently in CLI
+                new_body["loadTestConfiguration"]["quickStartTest"] = False
+                if data.get("quickStartTest"):
+                    logger.warning(
+                        "Quick start test is not supported currently in CLI. Please use portal to run quick start test"
+                    )
+                new_body["loadTestConfiguration"]["splitAllCSVs"] = False
+                if data.get("splitAllCSVs"):
+                    logger.warning(
+                        "CSV splitting is not supported currently in CLI. Please use portal to split CSVs"
+                    )
+                # implementation of failure criteria, secrets, cert and env var is pending
+        except (IOError, OSError) as ex:
+            if getattr(ex, "errno", 0) == errno.ENOENT:
+                raise ValidationError(f"{load_test_config_file} does not exist") from ex
+            raise
+        except (yaml.parser.ParserError, UnicodeDecodeError) as ex:
+            raise ValidationError(
+                f"Error parsing {load_test_config_file} ({str(ex)})"
+            ) from ex
+
+    else:
+        if display_name is not None:
+            new_body["displayName"] = display_name
+        else:
+            new_body["displayName"] = body.get("displayName", test_id)
+
+        if test_description is not None:
+            new_body["description"] = test_description
+        else:
```

Comment:
This can be done, but there are many cases where we need to handle them separately such as failureCriteria is only allowed through yaml for now, if both arguments and yaml is given we need to take yaml only with a proper warning message, etc. Therefore, for now we are going with this only.